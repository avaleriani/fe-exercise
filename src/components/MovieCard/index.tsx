import Link from "next/link";
import React, { FC } from "react";
import { formattedDate } from "../../../util/date";
import {
  ImageContainer,
  MovieCardWrapper,
  InfoContainer,
  LinkStyled,
  GenresWrapper,
  GenreStyled,
} from "./styled";

const MovieCard: FC<IMovieCard> = ({
  poster_path,
  title,
  genres,
  release_date,
  id,
}) => {
  return (
    <MovieCardWrapper data-testid="MovieCard__Component">
      <Link href={`/movie/${id}`}>
        <ImageContainer>
          <img src={poster_path} alt={title} />
        </ImageContainer>
      </Link>
      <InfoContainer>
        <LinkStyled href={`/movie/${id}`}>{title}</LinkStyled>
        <GenresWrapper>
          <GenreStyled>{genres?.toString()}</GenreStyled>
        </GenresWrapper>
        <div>{formattedDate(release_date)}</div>
      </InfoContainer>
    </MovieCardWrapper>
  );
};

export default MovieCard;
