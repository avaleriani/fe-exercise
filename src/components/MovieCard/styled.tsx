import Link from "next/link";
import styled from "styled-components";
import { size } from "../../../util/constants";

export const MovieCardWrapper = styled.div`
  margin: 10px;
  @media (min-width: ${size.mobileL}) {
    min-width: 250px;
    max-width: 250px;
  }
`;

export const ImageContainer = styled.a`
  width: 100%;
  border-radius: 10px;
  color: black;
  display: block;
  & > img {
    width: 100%;
    object-fit: contain;
    border-radius: 10px;
  }
`;

export const InfoContainer = styled.div`
  position: relative;
  padding: 15px 10px 10px 10px;
  color: white;
`;

export const LinkStyled = styled(Link)`
  white-space: nowrap;
  overflow: hidden;
  font-weight: bold;
  font-size: 18px;
  text-overflow: ellipsis;
  display: block;
`;

export const GenresWrapper = styled.div`
  margin: 16px 0;
`;

export const GenreStyled = styled.div`
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  font-size: 17px;
`;
