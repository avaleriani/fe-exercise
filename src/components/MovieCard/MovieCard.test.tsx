import { render, screen } from "@testing-library/react";
import MovieCard from "./index";

describe("MovieCard", () => {
  const movie = {
    adult: false,
    backdrop_path:
      "https://image.tmdb.org/t/p/w500/22z44LPkMyf5nyyXvv8qQLsbom.jpg",
    genres: ["Horror", "Mystery", "Thriller"],
    id: 631842,
    original_language: "en",
    original_title: "Knock at the Cabin",
    direction: "M. Night Shyamalan",
    overview:
      "While vacationing at a remote cabin, a young girl and her two fathers are taken hostage by four armed strangers who demand that the family make an unthinkable choice to avert the apocalypse. With limited access to the outside world, the family must decide what they believe before all is lost.",
    popularity: 4969.06,
    poster_path:
      "https://image.tmdb.org/t/p/w500/dm06L9pxDOL9jNSK4Cb6y139rrG.jpg",
    release_date: "2023-02-01",
    title: "Knock at the Cabin",
    video: false,
    vote_average: 6.6,
    vote_count: 778,
  };

  test("renders the movie title", () => {
    render(<MovieCard {...movie} />);
    const titleElement = screen.getByText(movie.title);
    expect(titleElement).toBeInTheDocument();
  });

  test("renders the movie poster image", () => {
    render(<MovieCard {...movie} />);
    const imageElement = screen.getByRole("img");
    expect(imageElement).toHaveAttribute("src", movie.poster_path);
    expect(imageElement).toHaveAttribute("alt", movie.title);
  });

  test("renders the movie genres", () => {
    render(<MovieCard {...movie} />);
    const genresElement = screen.getByText(movie.genres.join(", "));
    expect(genresElement).toBeInTheDocument();
  });

  test("renders the formatted release date", () => {
    render(<MovieCard {...movie} />);
    const dateElement = screen.getByText("February 1, 2023");
    expect(dateElement).toBeInTheDocument();
  });

  test("renders a link to the movie details page", () => {
    render(<MovieCard {...movie} />);
    const linkElement = screen.getByTestId("MovieBackground__Link");
    expect(linkElement).toBeInTheDocument();
  });

  it("renders character card unchanged", () => {
    const { container } = render(
      <MovieCard
        id={0}
        poster_path={""}
        title={""}
        genres={[]}
        release_date={""}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
