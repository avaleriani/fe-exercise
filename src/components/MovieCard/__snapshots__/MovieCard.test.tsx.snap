// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders character card unchanged 1`] = `
<div>
  <div
    class="sc-23eb2687-0 jOOGww"
    data-testid="MovieCard__Component"
  >
    <a
      data-testid="MovieBackground__Link"
      href="?movieId=0"
    >
      <div
        class="sc-23eb2687-1 hhdxOX"
      >
        <img
          alt=""
          src=""
        />
      </div>
    </a>
    <div
      class="sc-23eb2687-2 EETLY"
    >
      <a
        class="sc-23eb2687-3 Jkggr"
        href="?movieId=0"
      />
      <div
        class="sc-23eb2687-4 fSFKHU"
      >
        <div
          class="sc-23eb2687-5 beRdVS"
        />
      </div>
      <div />
    </div>
  </div>
</div>
`;
